src/resampling_function.rs

```diff
@@ -41,5 +57,25 @@ pub enum ResamplingFunction<
     Count,
     /// A custom resampling function that takes a closure that takes a slice of
     /// samples and returns an optional value.
-    Custom(CustomResamplingFunction<S, T>),
+    Custom(Box<dyn ClonableFnMut<S, T>>),
+}
+
+impl<
+        T: Div<Output = T> + std::iter::Sum + Default + Debug + Clone,
+        S: Sample<Value = T> + Clone,
+    > Clone for ResamplingFunction<T, S>
+{
+    fn clone(&self) -> Self {
+        match self {
+            Self::Average => Self::Average,
+            Self::Sum => Self::Sum,
+            Self::Max => Self::Max,
+            Self::Min => Self::Min,
+            Self::First => Self::First,
+            Self::Last => Self::Last,
+            Self::Coalesce => Self::Coalesce,
+            Self::Count => Self::Count,
+            Self::Custom(arg0) => Self::Custom(arg0.clone_box()),
```

Comment:
I was wondering about the name `arg0` of the function. Maybe you can give it a more descriptive name, e.g. `func` or just `f`.